packages/graphql-yoga/src/types.ts

```diff
@@ -0,0 +1,33 @@
+import type { BaseNodeGraphQLServerOptions } from '@graphql-yoga/core'
+import type { FastifyCorsOptions } from 'fastify-cors'
+import type { DocumentNode } from 'graphql'
+import type { IncomingHttpHeaders } from 'http'
+import type { OutgoingHttpHeaders } from 'http2'
+import type { TypedDocumentNode } from '@graphql-typed-document-node/core'
+
+/**
+ * Configuration options for the server
+ */
+export type GraphQLServerOptions = BaseNodeGraphQLServerOptions & {
+  cors?: FastifyCorsOptions
+  uploads?: boolean
+  /**
+   * Enable pino logging
+   * @default true
+   */
+  enableLogging?: boolean
+}
+
+export type GraphQLServerInject<
+  TData = any,
+  TVariables = Record<string, any>,
+> = {
+  /** GraphQL Operation to execute */
+  operation: string | DocumentNode | TypedDocumentNode<TData, TVariables>
```

Comment:
nitpicking: `document` might be the better term, as this string can contain multiple operation definitions 😇